Invalid Path in TabletGroupWatcher

The deleteRows test in ShellServerIT is failing due to recent change #2750 
<pre>
2022-06-09T10:42:04,829 [manager.EventCoordinator] INFO : Merge state of 1k;row7;row5 set to MERGING
2022-06-09T10:42:04,829 [manager.Manager] DEBUG: Deleting tablets for 1k;row7;row5
2022-06-09T10:42:04,830 [manager.Manager] DEBUG: Found following tablet 1k<;row7
2022-06-09T10:42:04,830 [manager.Manager] DEBUG: Making file deletion entries for 1k;row7;row5
2022-06-09T10:42:04,831 [manager.Manager] DEBUG: Finished gathering information from 1 of 1 servers in 0.00 seconds
2022-06-09T10:42:04,833 [manager.Manager] ERROR: Unable merge metadata table records
org.apache.accumulo.core.client.AccumuloException: java.lang.IllegalArgumentException: Invalid path provided, no scheme in t-000008q
        at org.apache.accumulo.manager.TabletGroupWatcher.deleteTablets(TabletGroupWatcher.java:694) ~[accumulo-manager-2.1.0-SNAPSHOT.jar:2.1.0-SNAPSHOT]
        at org.apache.accumulo.manager.TabletGroupWatcher.updateMergeState(TabletGroupWatcher.java:594) ~[accumulo-manager-2.1.0-SNAPSHOT.jar:2.1.0-SNAPSHOT]
        at org.apache.accumulo.manager.TabletGroupWatcher.run(TabletGroupWatcher.java:341) ~[accumulo-manager-2.1.0-SNAPSHOT.jar:2.1.0-SNAPSHOT]
Caused by: java.lang.IllegalArgumentException: Invalid path provided, no scheme in t-000008q
        at org.apache.accumulo.core.metadata.ValidationUtil.validate(ValidationUtil.java:47) ~[accumulo-core-2.1.0-SNAPSHOT.jar:2.1.0-SNAPSHOT]
        at org.apache.accumulo.core.metadata.ValidationUtil.validate(ValidationUtil.java:41) ~[accumulo-core-2.1.0-SNAPSHOT.jar:2.1.0-SNAPSHOT]
        at org.apache.accumulo.server.metadata.ServerAmpleImpl.createDeleteMutation(ServerAmpleImpl.java:208) ~[accumulo-server-base-2.1.0-SNAPSHOT.jar:2.1.0-SNAPSHOT]
        at org.apache.accumulo.server.metadata.ServerAmpleImpl.putGcFileAndDirCandidates(ServerAmpleImpl.java:136) ~[accumulo-server-base-2.1.0-SNAPSHOT.jar:2.1.0-SNAPSHOT]
        at org.apache.accumulo.manager.TabletGroupWatcher.deleteTablets(TabletGroupWatcher.java:666) ~[accumulo-manager-2.1.0-SNAPSHOT.jar:2.1.0-SNAPSHOT]
        ... 2 more
</pre>

--- a/core/src/main/java/org/apache/accumulo/core/gc/Reference.java
+++ b/core/src/main/java/org/apache/accumulo/core/gc/Reference.java
@@ -27,7 +27,6 @@ import org.apache.accumulo.core.metadata.schema.MetadataSchema;
 public class Reference implements Comparable<Reference> {
   // parts of an absolute URI, like "hdfs://1.2.3.4/accumulo/tables/2a/t-0003"
   public final TableId tableId; // 2a
-  public final String tabletDir; // t-0003
 
   // the exact string that is stored in the metadata
   public final String metadataEntry;
@@ -36,7 +35,6 @@ public class Reference implements Comparable<Reference> {
     MetadataSchema.TabletsSection.ServerColumnFamily.validateDirCol(tableId.canonical());
     this.tableId = tableId;
     this.metadataEntry = metadataEntry;
-    this.tabletDir = metadataEntry;
   }
 
   @Override

--- a/core/src/main/java/org/apache/accumulo/core/gc/ReferenceDirectory.java
+++ b/core/src/main/java/org/apache/accumulo/core/gc/ReferenceDirectory.java
@@ -24,8 +24,10 @@ import org.apache.accumulo.core.data.TableId;
  * Part of the Tablet File path that is definitely a directory.
  */
 public class ReferenceDirectory extends Reference {
+  public final String tabletDir; // t-0003
 
   public ReferenceDirectory(TableId tableId, String dirName) {
     super(tableId, dirName);
+    this.tabletDir = dirName;
   }
 }

--- a/server/manager/src/main/java/org/apache/accumulo/manager/TabletGroupWatcher.java
+++ b/server/manager/src/main/java/org/apache/accumulo/manager/TabletGroupWatcher.java
@@ -53,7 +53,6 @@ import org.apache.accumulo.core.data.TableId;
 import org.apache.accumulo.core.data.Value;
 import org.apache.accumulo.core.dataImpl.KeyExtent;
 import org.apache.accumulo.core.gc.Reference;
-import org.apache.accumulo.core.gc.ReferenceDirectory;
 import org.apache.accumulo.core.logging.TabletLogger;
 import org.apache.accumulo.core.manager.state.tables.TableState;
 import org.apache.accumulo.core.manager.thrift.ManagerState;
@@ -642,8 +641,7 @@ abstract class TabletGroupWatcher extends AccumuloDaemonThread {
         Key key = entry.getKey();
         if (key.compareColumnFamily(DataFileColumnFamily.NAME) == 0) {
           var stf = new StoredTabletFile(key.getColumnQualifierData().toString());
-          var tabletDirectory = new ReferenceDirectory(stf.getTableId(), stf.getTabletDir());
-          datafilesAndDirs.add(tabletDirectory);
+          datafilesAndDirs.add(new Reference(stf.getTableId(), stf.getMetaUpdateDelete()));
           if (datafilesAndDirs.size() > 1000) {
             ample.putGcFileAndDirCandidates(extent.tableId(), datafilesAndDirs);
             datafilesAndDirs.clear();

--- a/server/manager/src/test/java/org/apache/accumulo/manager/upgrade/Upgrader9to10Test.java
+++ b/server/manager/src/test/java/org/apache/accumulo/manager/upgrade/Upgrader9to10Test.java
@@ -98,7 +98,7 @@ public class Upgrader9to10Test {
 
   private void compareReferences(Reference ref1, Reference ref2) {
     assertEquals(ref1.metadataEntry, ref2.metadataEntry);
-    assertEquals(ref1.tabletDir, ref2.tabletDir);
+    assertEquals(ref1.tableId, ref2.tableId);
   }
 
   @Test

